fs/cache/cache.go

```diff
@@ -99,7 +100,9 @@ func GetFn(ctx context.Context, fsString string, create func(ctx context.Context
 // Pin f into the cache until Unpin is called
 func Pin(f fs.Fs) {
 	createOnFirstUse()
-	c.Pin(fs.ConfigString(f))
+	configString := fs.ConfigString(f)
+	c.Pin(configString)
+	c.PinPrefix(strings.TrimSuffix(configString, "/")) // also pin its aliases, if any
```

Comment:
I'm worried that c.PinPrefix would pin things it shouldn't here, so if you made `remote:path/dir` then made `remote/path/file.txt` then the latter would pin the former.

Reply:
I think this would also be true of [`c.DeletePrefix`](https://github.com/rclone/rclone/blob/b7422a4fc8b747d18c01834bf5a14f1a92ca596c/fs/cache/cache.go#L189) which is where I got the idea, and there are probably ways to refine the targeting, but it's probably moot now as my proposed revision doesn't have a `PinPrefix`.